subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/SubstIntegrationTest.groovy

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.integtests
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import org.gradle.util.TextUtil
+
+@Requires(TestPrecondition.WINDOWS)
+class SubstIntegrationTest extends AbstractIntegrationSpec {
+    def "up to date check works from filesystem's root - intput folder to output file"() {
+        def drive = 'X:'.toUpperCase()
```

Comment:
Why the call to `toUpperCase`?

Reply:
Since I need it to be uppercase in https://github.com/gradle/gradle/pull/9592/files#diff-ac225a0c7c15b7f81800a3a841aab1a6R107

I could of course uppercase it there as well, just wanted to standardize it as early as possible.
(thought about someone changing it to something lowercase, thinking that it shouldn't matter on Windows and breaking the code, since it's sometimes handled as a String).
Would you like me to uppercase it only where it matters, i.e. `isPresent`?

Reply:
I think the method should be agnostic to whether the drive letter is uppercase or not. So I wouldn't require clients of the methods which take a drive letter to upper case it before hand. And it looks very strange calling `toUpperCase` on something which is already upper case.

So either document that the method(s) require upper case letters and drop the `toUpperCase()` here or (better) make the methods general enough so that they accept arbitrary strings.